src/timesync/timesyncd-manager.c

```diff
@@ -383,10 +383,10 @@ static int manager_adjust_clock(Manager *m, double offset, int leap_sec) {
                   "  offset       : %+.3f sec\n"
                   "  freq offset  : %+li (%i ppm)\n",
                   tmx.status, tmx.status & STA_UNSYNC ? "unsync" : "sync",
-                  tmx.time.tv_sec, tmx.time.tv_usec / NSEC_PER_MSEC,
-                  tmx.constant,
+                  (long int)tmx.time.tv_sec, tmx.time.tv_usec / NSEC_PER_MSEC,
+                  (long int)tmx.constant,
                   (double)tmx.offset / NSEC_PER_SEC,
-                  tmx.freq, m->drift_ppm);
+                  (long int)tmx.freq, m->drift_ppm);
```

Comment:
That seems wrong too. If I understand what you wrote above correctly, this cast reduce the size of the variable from 64 bits to 32 bits. We shouldn't do do reductions like that. That type is `__kernel_long_t`, so maybe it's necessary to add PRI_KERNEL_LONG_T in print-format.h and use it as appropriate.